webhook/webhook_test.go

```diff
@@ -138,6 +138,68 @@ func TestValidCreateResourceSucceedsWithDefaultPatch(t *testing.T) {
 	})
 }
 
+func TestValidCreateResourceSucceedsWithRoundTripAndDefaultPatch(t *testing.T) {
+	req := &admissionv1beta1.AdmissionRequest{
+		Operation: admissionv1beta1.Create,
+		Kind: metav1.GroupVersionKind{
+			Group:   "pkg.knative.dev",
+			Version: "v1alpha1",
+			Kind:    "InnerDefaultResource",
+		},
+	}
+
+	b := createInnerDefaultResourceWithoutSpec()
+	req.Object.Raw = b
+
+	_, ac := newNonRunningTestAdmissionController(t, newDefaultOptions())
+	resp := ac.admit(TestContextWithLogger(t), req)
+	expectAllowed(t, resp)
+	expectPatches(t, resp.Patch, []jsonpatch.JsonPatchOperation{
+		{
+			Operation: "add",
+			Path:      "/spec",
+			Value:     map[string]interface{}{},
+		},
+		// This is almost identical to incrementGenerationPatch(0), but uses 'add', rather than
+		// 'replace' because `spec` is empty to begin with.
+		{
+			Operation: "add",
+			Path:      "/spec/generation",
+			Value:     float64(1),
+		},
+		{
+			Operation: "add",
+			Path:      "/spec/fieldWithDefault",
+			Value:     "I'm a default.",
+		},
+	})
+}
+
+func createInnerDefaultResourceWithoutSpec() []byte {
+	r := InnerDefaultResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNamespace,
+			Name:      "a name",
+		},
+	}
+	// Remove the 'spec' field of the generated JSON by marshaling it to JSON, parsing that as a
+	// generic map[string]interface{}, removing 'spec', and marshaling it again.
+	origBytes, err := json.Marshal(r)
+	if err != nil {
+		panic(err)
```

Comment:
please don't panic in tests.
Return error and then fatal in the test.

Reply:
Done. Out of curiosity, why no panics during tests?

Reply:
I'm guessing because the panics blocks other tests from running, so if this test breaks then you don't get the status of the others.